### PR TITLE
Make contributing notes much more helpful

### PR DESCRIPTION
**Pull request by davidhewitt**
I get the impression that we have a number of potential contributors waiting in the wings, so I just wanted to make it easier for them to find out how to join in.

### CODE REVIEW

**Comment by davidhewitt** [comment]
(I went through the existing PyO3 issues and labelled many with [needs-implementer](https://github.com/PyO3/pyo3/issues?q=is%3Aissue+is%3Aopen+label%3Aneeds-implemeter) and [needs-design](https://github.com/PyO3/pyo3/issues?q=is%3Aissue+is%3Aopen+label%3Aneeds-design), which I think was a useful thing to do. I am happy to bikeshed on the label names if people have other preferences. This pair, while not perfect, seemed good enough to vaguely label things for people to discover what's needed.)

File: Contributing.md [hunk 1]
```diff
@@ -1,14 +1,66 @@
...
+
+### Dive straight into the implementation
+
+We use the [needs-implementer](https://github.com/PyO3/pyo3/issues?q=is%3Aissue+is%3Aopen+label%3Aneeds-implemeter) label to mark issues where the solution is reasonably clear and nobody has yet claimed responsibility for its enaction.
+
+Feel free to introduce yourself directly on any issue which strikes your interest and volunteer to take it on!
+
+Don't be afraid if the full solution is not clear to you! The core PyO3 contributors will be happy to mentor you through any questions you have to help you write the solution.
+
+### Help write great docs
```
Comment (kngwyu) [inline Contributing.md:27]: Let's note that we have guides and API docs.

**Comment by kngwyu** [comment]
Thank you!

I think that this documentation is well written
However, how about making it simpler?
Now many people use and have interests in PyO3, including lots of non-native English speakers.
For them, I think it's important that the document is clearly structured since they may not want to read many English sentences.

I recommend the following section structure, based on the [Rustc dev guide](https://rustc-dev-guide.rust-lang.org/getting-started.html): 
```markdown
## Asking Questions
- Note that we have gitter channel and Github issue can also be used when the bug is clear.
## Sending Pull requests
- Note some tips as PR templates. Maybe checklist is good.
- Note how to request reviews.
## How to discover starting points
- List some issue labels as you write
```

Also, I feel some phrasing difficult and recommend simpler words (e.g., I feel it difficult to understand 'nuts and bolts' and 'development is coordinated'). Maybe [easy_rust](https://github.com/Dhghomon/easy_rust) is helpful for wording.

**Comment by davidhewitt** [comment]
> Also, I feel some phrasing difficult and recommend simpler words (e.g., I feel it difficult to understand 'nuts and bolts' and 'development is coordinated'). Maybe easy_rust is helpful for wording.

Thanks, I think you're right I probably made it too colourful. I wanted to make it fun because I think contributing to pyo3 is exciting!

I'll push a rewording later which simplifies the most exotic phrasing.

**Comment by davidhewitt** [comment]
I've made a few changes which hopefully make this easier to read. I think it's better now; thanks for the feedback!

**Pull request merged**